common/src/chain/pow.rs

```diff
@@ -91,21 +91,21 @@ const fn allow_min_difficulty_blocks(chain_type: ChainType) -> bool {
     }
 }
 
-pub(crate) const fn limit(chain_type: ChainType) -> Uint256 {
+pub(crate) const fn limit(chain_type: ChainType) -> U256 {
     match chain_type {
-        ChainType::Mainnet | ChainType::Testnet => Uint256([
+        ChainType::Mainnet | ChainType::Testnet => U256::from_words([
```

Comment:
Using "from words" is wrong. This basically means mintlayer-core will never compile on 32-bit systems, because the size of a word in 32-bit systems is 32-bit.

Please use the hex versions so that it doesn't depend on word size.